pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -455,3 +460,31 @@ func fixLibvirtCrcNetworkActive() (bool, error) {
 	}
 	return true, nil
 }
+
+func checkCrcDnsmasqConfigFile() (bool, error) {
+	c := []byte(crcDnsmasqConfig)
+	_, err := os.Stat(crcDnsmasqConfigPath)
+	if err != nil {
+		return false, fmt.Errorf("File not found: %s: %s", crcDnsmasqConfigPath, err.Error())
+	}
+	config, err := ioutil.ReadFile(crcDnsmasqConfigPath)
+	if err != nil {
+		return false, fmt.Errorf("Error opening file: %s: %s", crcDnsmasqConfigPath, err.Error())
+	}
+	if !bytes.Equal(config, c) {
+		return false, fmt.Errorf("Config file contains changes: %s: %s", crcDnsmasqConfigPath)
+	}
+	return true, nil
+}
+
+func fixCrcDnsmasqConfigFile() (bool, error) {
+	cmd := exec.Command("sudo", "tee", crcDnsmasqConfigPath)
+	cmd.Stdin = strings.NewReader(crcDnsmasqConfig)
+	buf := new(bytes.Buffer)
+	cmd.Stderr = buf
+	err := cmd.Run()
+	if err != nil {
+		return false, fmt.Errorf("Failed to write dnsmasq config file: %s: %s: %v", crcDnsmasqConfigPath, buf.String(), err)
+	}
+	return true, err
```

Comment:
@gbraad Problem is just adding or changing the file content not going to work until you reload the NetworkManager service which should be part of this function.

Reply:
will add a shared SystemD control function like we used in minishift

Reply:
Done